nodeup/pkg/model/kube_apiserver.go

```diff
@@ -287,6 +287,8 @@ func (b *KubeAPIServerBuilder) writeAuthenticationConfig(c *fi.ModelBuilderConte
 // buildPod is responsible for generating the kube-apiserver pod and thus manifest file
 func (b *KubeAPIServerBuilder) buildPod() (*v1.Pod, error) {
 	kubeAPIServer := b.Cluster.Spec.KubeAPIServer
+	// TODO pass the public key instead. We would first need to segregate the secrets better.
```

Comment:
Yeah, this would be great.  I think we've started doing things under `/etc/kubernetes`; the two suggestions I've heard/seen are `/etc/kubernetes/secrets/<component>` and `/etc/kubernetes/pki/<component>`  Or maybe `/etc/kubernetes/<component>/secrets` :-)  Regardless of the naming nit-pick, I think splitting by component would be good, particularly in conjunction with the idea you're championing of having things run under dedicated non-root unix accounts.

Reply:
I think we want `/etc/kubernetes/<component>/` with nothing mounting `/etc/kubernetes/`